cristin-project/src/main/java/no/unit/nva/cristin/projects/common/QueryBuilderProject.java

```diff
@@ -0,0 +1,456 @@
+                    query.setValue(key, PARAMETER_PAGE_DEFAULT_VALUE);
+                    break;
+                case PAGE_ITEMS_PER_PAGE:
+                    query.setValue(key, PARAMETER_PER_PAGE_DEFAULT_VALUE);
+                    break;
+                default:
+                    break;
+            }
+        });
+    }
+
+    @Override
+    protected void setPath(String key, String value) {
+        asNvaQuery();
+        var nonNullValue = nonNull(value) ? value : EMPTY_STRING;
+
+        if (key.equals(PATH_IDENTITY.getNvaKey())) {
+            withPathIdentity(nonNullValue);
+        } else if (key.equals(PATH_ORGANISATION.getNvaKey())) {
+            withPathOrganization(nonNullValue);
+        } else if (key.equals(PATH_PROJECT.getNvaKey()) || key.equals(PATH_PROJECT.getKey())) {
+            withPathProject(nonNullValue);
+        } else {
+            invalidKeys.add(key);
+        }
+    }
+
+    @Override
+    protected void setValue(String key, String value) {
+        var qpKey = keyFromString(key,value);
+        if (!key.equals(qpKey.getKey()) && !query.isNvaQuery() && qpKey != INVALID) {
+            asNvaQuery();
+        }
+        switch (qpKey) {
+            case PATH_IDENTITY:
+            case PATH_PROJECT:
+                withPathIdentity(value);
+                break;
+            case PATH_ORGANISATION:
+                withPathOrganization(value);
+                break;
+            case ORGANIZATION:
+                withOrganization(value);
+                break;
+            case BIOBANK:
+                withBiobank(value);
+                break;
+            case KEYWORD:
+                withKeyword(value);
+                break;
+            case PARTICIPANT:
+                withParticipant(value);
+                break;
+            case QUERY:
+                withQuery(value);
+                break;
+            case STATUS:
+                withStatus(value);
+                break;
+            case APPROVAL_REFERENCE_ID:
+            case APPROVED_BY:
+            case FUNDING:
+            case FUNDING_SOURCE:
+            case GRANT_ID:
+            case INSTITUTION:
+            case LANGUAGE:
+            case LEVELS:
+            case MODIFIED_SINCE:
+            case NAME:
+            case PROJECT_MANAGER:
+            case PROJECT_UNIT:
+            case TITLE:
+            case USER:
+            case PAGE_CURRENT:
+            case PAGE_ITEMS_PER_PAGE:
+            case PAGE_SORT:
+                query.setValue(qpKey, value);
+                break;
+            default:
+                invalidKeys.add(key);
+                break;
+        }
+    }
+
+    @Override
+    protected Set<String> validKeys() {
+        return query.isNvaQuery() ? VALID_QUERY_PARAMETER_NVA_KEYS : VALID_QUERY_PARAMETER_KEYS;
+    }
+
+    @Override
+    protected void throwInvalidParamererValue(Entry<ParameterKeyProject, String> entry) throws BadRequestException {
```

Comment:
a lot of duplicate as in QueryBuilder but maybe cannot be avoided

Reply:
Don't understand what you are referring to here?